ping_server.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+ 
+# -q quiet
+# -c number of pings to perform
+FILE="/var/log/ping_server.log"
+ 
+ping -q -c2 opinion.berkeley.edu > /dev/null
+ 
+if [ $? -eq 0 ]
+then
+    echo "ok"
+else
+    DATE=`date '+%Y-%m-%d %H:%M:%S'`
+    sudo echo "Server down at ${DATE}" > $FILE
```

Comment:
Use `>> $FILE` rather than `> $FILE`--`>` overwrites the entire file, while `>>` appends input to the file (i.e. we want to detect more than the latest outage).